doc/services/shell/index.rst

```diff
@@ -74,6 +74,28 @@ procedure:
   to the shell.
 
 
+Telnet Backend
+==============
+
+Enabling :kconfig:option:`CONFIG_SHELL_BACKEND_TELNET` will allow users to use telnet
+as a shell backend. Connecting to it can be done using PuTTY or any ``telnet`` client.
+For example:
+
+.. code-block:: none
+
+  telnet <ip address> <port>
+
+By default the telnet client won't handle telnet commands. Although command support
+can be enabled with :kconfig:option:`CONFIG_SHELL_TELNET_SUPPORT_COMMAND`. This will
+give the telnet client access to a very limited set of supported commands but still
+can be turned on if needed. One of the command option it supports is the ``ECHO``
+option. This will allow the client to be in character mode (character at a time),
+similar to a uart backend in that regard. This will make the client send a character
+as soon as it is typed having the effect of increasing the network traffic
+considerably. Although it will give the user the ability to use the keyboard arrows
+as well as the DELETE, HOME, END, INSERT keyboard keys.
```

Comment:
Please fix: I think it would be better to refer to the features at a higher level, including [internal cross-reference links](https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#internal-cross-reference-linking):
```suggestion
considerably. For that cost, it will enable the line editing, `tab completion <tab-feature_>`, and `history <history-feature>_` features of the shell.
```

Reply:
Very good idea